src/json.hpp.re2c

```diff
@@ -800,6 +800,11 @@ void to_json(BasicJsonType& j, const  CompatibleObjectType& arr)
     external_constructor<value_t::object>::construct(j, arr);
 }
 
+template <typename BasicJsonType, typename T, std::size_t N, enable_if_t<not std::is_constructible<typename BasicJsonType::string_t, T(&)[N]>::value, int> = 0>
+void to_json(BasicJsonType &j, T (&tab)[N]) {
```

Comment:
We could exploit the knowledge of the size and call a reserve on the internal vector before.